### DISCUSSION TITLE
i think this _isApprovedOrOwner is no more in openzeppelin

### DISCUSSION THREAD

**Discussion by king-null** [Cyfrin/foundry-full-course-cu#1592] [Q&A]
LESSON 11 section 15 , _isApprovedOrOwner is giving an error undeclared identifier, so i decided to search in openzeppelin and still couldnt find the function in the ERC721 so i think is been scrapped out or something please if anyone know the function in open zeppelin i can use to replace this _isApprovedOrOwner function 

below is my flipMood function

```solidity
 function flipMood(uint256 tokenId) public {
        // only want the NFT owner to be able to change the mood
        // and the ERC721 from openzeppelin come with a fuction called _isApprovedOrOwner
        if (!_isApprovedOrOwner(msg.sender, tokenId)) {
            revert MoodNft__CantFlipMoodIfNotOwner();
        }
        if (s_tokenIdToMood[tokenId] == Mood.HAPPY) {
            s_tokenIdToMood[tokenId] = Mood.SAD;
        } else {
            s_tokenIdToMood[tokenId] = Mood.HAPPY;
        }
    }
```

**Comment by EngrPips** [comment]
Hello @king-null , you should always check the `chronological-updates.md` on the github repo associated to the course if something seems to be different from what is in the course as most of the times it would have been written in the `chronological-updates.md` and the solution like shown below 

<img width="1440" alt="Screenshot 2024-04-11 at 3 03 24 AM" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/95760918/8e69c230-680f-4420-b131-b4db9b912098">

`https://github.com/Cyfrin/foundry-full-course-f23/blob/main/chronological-updates.md`

**Reply by FrankezeCode** [reply]
> Hello @king-null , you should always check the `chronological-updates.md` on the github repo associated to the course if something seems to as most of the times it would have been written in the `chronological-updates.md` and the solution like shown below
> 
> <img alt="Screenshot 2024-04-11 at 3 03 24 AM" width="1440" src="https://private-user-images.githubusercontent.com/95760918/321590337-8e69c230-680f-4420-b131-b4db9b912098.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE5MTM1NzEsIm5iZiI6MTcyMTkxMzI3MSwicGF0aCI6Ii85NTc2MDkxOC8zMjE1OTAzMzctOGU2OWMyMzAtNjgwZi00NDIwLWIxMzEtYjRkYjliOTEyMDk4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzI1VDEzMTQzMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWJjMjQyMThkNTM1NzIxZGY1YTNmNDMzNjIyZDVkODNjNWUyYzY3MTRmMTA1OGEwMzhhNGE3MDkyNGIxYTY1MWYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.ZF6WBCWvsPdZzjktW9UXsrXv--OM11Z69o1j6H8Havs">
> `https://github.com/Cyfrin/foundry-full-course-f23/blob/main/chronological-updates.md`

@EngrPips  thanks you are awesome 💯,  I didn't know about this earlier

**Reply by EngrPips** [reply]
You are welcome, Friend.

**Reply by wilson0922** [reply]
I also need this, thank you all~
